### DISCUSSION TITLE
'Unsupported' Error on 2.5.303 & NodeJS 22.4.0

### DISCUSSION THREAD

**Discussion by Heirys510** [requarks/wiki#7250] [Error / Bug Report]
Hello,

we are seeing the below after upgrading to 2.5.303 & NJS 22.4.0?

2024-07-05T13:06:14.081Z [MASTER] info: HTTP Server on port: [ 3000 ]
2024-07-05T13:06:14.128Z [MASTER] info: HTTPS Server on port: [ 443 ]
2024-07-05T13:06:14.170Z [MASTER] error: unsupported

Have also tried this on 2.5.300 NJS 18.x or 20.x to the same result

**Comment by tw-auh** [comment]
I've got the same problem, using node v20.16.0 and wiki.js Version 2.5.303.
I have to mention that I get this on a migrated install, so maybe my problem doesn't match a fresh install.

EDIT: I got this up and running with a downgrade to node v 16 which is out of support. So I would also prefer to upgrade to a stable version.

**Reply by Heirys510** [reply]
Yeah we had the same experience - downgrading to v16 works however is out of support and we are required to upgrade past V16 for CVE-2024-27980.

Have imaged a fresh Server2022 VM and experience the same behaviour.

**Reply by Heirys510** [reply]
Any news on this one? 
We still are unable to upgrade past v16 without the Service going down.

**Comment by mihailim** [comment]
Not seeing this on Debian stable at least, with the distribution-shipped 18.19.0+dfsg-6~deb12u2. What OS are you using?

**Reply by Heirys510** [reply]
Behaviour appears on both Server2016 & 2022

**Comment by tw-auh** [comment]
I recently encountered issues with Wiki.js after performing a release upgrade on my Ubuntu server (from 22.04 to 24.04). The upgrade also updated Node.js to version 18, which broke Wiki.js. I wanted to share how I resolved this and successfully got it running with Node.js v22.

Initially, I tried cleaning the npm packages and performing a fresh npm install, but this didn’t solve the issue. So I decided to take a different approach:

- Backup: I backed up my config.yml file to preserve my settings.
- Remove old installation: I completely removed the existing Wiki.js installation and Node.js.
- Install Node.js v22: I installed Node.js v22 from the official sources.
- Reinstall Wiki.js: I installed Wiki.js as usual, following the standard procedure.
- Restore the config: Finally, I copied my backed-up config.yml back into the installation folder.

After completing these steps, my Wiki.js instance was up and running again without issues.

I hope this helps anyone experiencing similar problems!
